core/state/statedb.go

```diff
@@ -1197,10 +1197,9 @@ func (s *StateDB) Prepare(rules params.Rules, sender, coinbase common.Address, d
 }
 
 // preparePredicateStorageSlots populates the predicateStorageSlots field from the transaction's access list
-// Note: if an address is specified multiple times in the access list, only the last storage slots provided
-// for it are used in predicates.
-// During predicate verification, we require that a precompile address is only specififed in the access list
-// once to avoid a situation where we verify multiple predicate and only expose data from the last one.
+// Note: if an address is specified multiple times in the access list, each storage slot for that address is
+// appended to a slice of byte slices. Each byte slice represents a predicate, making it a slice of predicates
+// for each access list address, and every predicate in the slice goes through verification.
 func (s *StateDB) preparePredicateStorageSlots(rules params.Rules, list types.AccessList) {
```

Comment:
noticed that the predicate check https://github.com/ava-labs/subnet-evm/blob/master/core/predicate_check.go#L35C47-L35C47 composing of `predicateArguments` is pretty similar to `preparePredicateStorageSlots`, not sure if any use in consolidating the two.

Reply:
Yup great find, I think this would make sense to add as a helper in https://github.com/ava-labs/subnet-evm/tree/master/utils/predicate. Do you want to make a PR to refactor this so that these two both call that helper?